lib/SILGen/SILGenPoly.cpp

```diff
@@ -399,6 +399,12 @@ ManagedValue Transform::transform(ManagedValue v,
     inputOTK = OTK_None;
   }
 
+  if (inputOTK != OTK_None && outputOTK == OTK_None) {
+    SILValue result = SGF.B.createUncheckedBitCast(Loc, v.getValue(),
```

Comment:
`assert` is firing in this function, because requirement is `Optional`, and witness isn't. 

I'd love to avoid ending up here :D